### PR TITLE
Use standard bit builtins instead of compiler extensions

### PR DESCRIPTION
**Pull request by ISSOtm**

### CODE REVIEW

File: src/fix/main.cpp [hunk 1]
```diff
@@ -1041,13 +1043,12 @@ static void processFile(int input, int output, char const *name, off_t fileSize)
...
 		}
 		assert(nbBanks >= 2);
+		// The subtraction would be invalid for 0, but the assertion above means it won't happen.
+		auto logNbBanks = std::bit_width(nbBanks - 1);
 		// Alter number of banks to reflect required value
-		// x&(x-1) is zero iff x is a power of 2, or 0; we know for sure it's non-zero,
-		// so this is true (non-zero) when we don't have a power of 2
-		if (nbBanks & (nbBanks - 1))
-			nbBanks = 1 << (CHAR_BIT * sizeof(nbBanks) - clz(nbBanks));
+		nbBanks = uint32_t(1) << logNbBanks;
```
Comment (Rangi42) [inline src/fix/main.cpp:1049]: Everywhere else uses C-style casts; we could consider converting those to constructors separately.
```suggestion
		nbBanks = (uint32_t)1 << logNbBanks;
```

File: src/fix/main.cpp [hunk 2]
```diff
@@ -1041,13 +1043,12 @@ static void processFile(int input, int output, char const *name, off_t fileSize)
 			assert(rom0Len == sizeof(rom0));
 		}
 		assert(nbBanks >= 2);
+		// The subtraction would be invalid for 0, but the assertion above means it won't happen.
+		auto logNbBanks = std::bit_width(nbBanks - 1);
```
Comment (Rangi42) [inline src/fix/main.cpp:1047]: It definitely returns an [`int`](https://en.cppreference.com/w/cpp/numeric/bit_width).
```suggestion
		int logNbBanks = std::bit_width(nbBanks - 1);
```

**Review by Rangi42 — CHANGES_REQUESTED** [review]
Nice, thank you! Always good to see LoC go down. :)

**Comment by Rangi42** [comment]
Ubuntu 20.04's g++ 9.3.0 doesn't support `std::bit_width`. :(

**Comment by ISSOtm** [comment]
Yeah, that sucks. How should we detect it, though? :/

**Comment by Rangi42** [comment]
We could directly check the `__GNUC__ ` version?

**Comment by ISSOtm** [comment]
Honestly, at that point we'd be back to square one. Let's not; we can revive this PR later, after updating our requirements.

**Pull request closed without merging**